Compilation fail with error: ‘bool’ cannot be defined via ‘typedef’

Apparently C23 introduces the true, false, and bool keywords, so LovyanGFX can fail to compile on systems with a very new GCC:

```
In file included from .pio/libdeps/native-mui/LovyanGFX/src/lgfx/utility/lgfx_qrcode.c:38:
.pio/libdeps/native-mui/LovyanGFX/src/lgfx/utility/lgfx_qrcode.h:42:23: error: ‘bool’ cannot be defined via ‘typedef’
   42 | typedef unsigned char bool;
      |                       ^~~~
.pio/libdeps/native-mui/LovyanGFX/src/lgfx/utility/lgfx_qrcode.h:42:23: note: ‘bool’ is a keyword with ‘-std=c23’ onwards
.pio/libdeps/native-mui/LovyanGFX/src/lgfx/utility/lgfx_qrcode.h:43:19: error: expected identifier or ‘(’ before ‘false’
   43 | static const bool false = 0;
      |                   ^~~~~
.pio/libdeps/native-mui/LovyanGFX/src/lgfx/utility/lgfx_qrcode.h:44:19: error: expected identifier or ‘(’ before ‘true’
   44 | static const bool true = 1;
      |                   ^~~~

```

--- a/platformio.ini
+++ b/platformio.ini
@@ -13,7 +13,7 @@ default_envs = native-mui
 
 [device-ui_base]
 lib_deps =
-  https://github.com/meshtastic/device-ui/archive/7656d49ea2bdb81a43afc1e8552f1cc86b8dce3c.zip
+  https://github.com/meshtastic/device-ui/archive/272defcb35651461830ebfd1b39c9167c8f49317.zip
   nanopb/Nanopb@^0.4.91
 
 
@@ -24,7 +24,7 @@ upload_protocol = esp-builtin
 board_build.filesystem = littlefs
 extra_scripts = ./bin/littlefsbuilder.py
 build_type = release
-build_flags = -std=c++17 -Os
+build_flags = -std=c++17 -std=gnu17 -Os
   -DCORE_DEBUG_LEVEL=ARDUHAL_LOG_LEVEL_VERBOSE
   -DLOG_LOCAL_LEVEL=ESP_LOG_VERBOSE
   -DCONFIG_ARDUHAL_ESP_LOG

